src/bus.rs

```diff
@@ -36,3 +57,69 @@ impl SharedMessageBus {
             .clone()
     }
 }
+
+#[tokio::test]
```

Comment:
I think it's a good practice to have tests in a specific mod block, compiled only when running cargo tests: 
```
#[cfg(test)]
mod test {

// tests here

}
```